app/controllers/api/lesson_completions_controller.rb

```diff
@@ -0,0 +1,28 @@
+module Api
+  class LessonCompletionsController < ApplicationController
+    before_action :authenticate
+
+    def index
+      render json: serialized_lesson_completions
+    end
+
+    private
+
+    def serialized_lesson_completions
+      Course.all.map do |course|
+        CourseSerializer.as_json(course, days)
+      end
+    end
+
+    def days
+      params[:days].to_i
+    end
+
+    def authenticate
+      authenticate_or_request_with_http_basic do |username, password|
+        username == ENV.fetch('API_USERNAME')
```

Comment:
@KevinMulhern Shouldn't this be `username == ENV.fetch('API_USERNAME') && password == ENV.fetch('API_PASSWORD')`? :thinking:

Reply:
At its current state, the code only checks for the `password` and not the combination of `username` and `password`.

Reply:
So it is, well spotted.